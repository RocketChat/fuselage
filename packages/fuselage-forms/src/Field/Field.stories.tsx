import type { StoryFn, Meta } from '@storybook/react-webpack5';

import {
  TextInput,
  EmailInput,
  PasswordInput,
  SearchInput,
  TextAreaInput,
  Select,
  CheckBox,
  ToggleSwitch,
  RadioButton,
  TelephoneInput,
  NumberInput,
  UrlInput,
} from '../Inputs';

import {
  Field,
  FieldDescription,
  FieldError,
  FieldHint,
  FieldLabel,
  FieldLabelInfo,
  FieldLink,
  FieldRow,
} from '.';

export default {
  title: 'Inputs/Field',
  component: Field,
} satisfies Meta<typeof Field>;

export const WithTextInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Text Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>
      Text inputs are used to enter a single line of text.
    </FieldDescription>
    <FieldRow>
      <TextInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithEmailInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Email Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>
      This field requires a valid email address
    </FieldDescription>
    <FieldRow>
      <EmailInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithPasswordInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Password Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>This field requires a valid password</FieldDescription>
    <FieldRow>
      <PasswordInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithSearchInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Search Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>
      You can use this field to search for things
    </FieldDescription>
    <FieldRow>
      <SearchInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithTextArea: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Text Area
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>
      You can use this field to enter multi-line text
    </FieldDescription>
    <FieldRow>
      <TextAreaInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithRadioButton: StoryFn<typeof Field> = () => (
  <Field>
    <FieldRow>
      <FieldLabel required>
        Example Radio Button
        <FieldLabelInfo title='with extra info in a tooltip' />
      </FieldLabel>
      <RadioButton />
    </FieldRow>
    <FieldDescription>You can only select a single option</FieldDescription>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithToggleSwitch: StoryFn<typeof Field> = () => (
  <Field>
    <FieldRow>
      <FieldLabel required>
        Example Toggle Switch
        <FieldLabelInfo title='with extra info in a tooltip' />
      </FieldLabel>
      <ToggleSwitch />
    </FieldRow>
    <FieldDescription>This field represents a boolean value</FieldDescription>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithCheckbox: StoryFn<typeof Field> = () => (
  <Field>
    <FieldRow>
      <FieldLabel required>
        Example Checkbox
        <FieldLabelInfo title='with extra info in a tooltip' />
      </FieldLabel>
      <CheckBox />
    </FieldRow>
    <FieldDescription>You can select multiple options</FieldDescription>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithTelephoneInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Telephone Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>
      This field requires a valid telephone number
    </FieldDescription>
    <FieldRow>
      <TelephoneInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
export const WithNumberInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example Number Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>This field requires a valid number</FieldDescription>
    <FieldRow>
      <NumberInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);

export const WithUrlInput: StoryFn<typeof Field> = () => (
  <Field>
    <FieldLabel required>
      Example URL Input
      <FieldLabelInfo title='with extra info in a tooltip' />
    </FieldLabel>
    <FieldDescription>This field requires a valid URL</FieldDescription>
    <FieldRow>
      <UrlInput />
    </FieldRow>
    <FieldError>You failed to enter a valid value</FieldError>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);

export const WithSelect: StoryFn<typeof Field> = () => (
  <Field>
    <FieldRow>
      <FieldLabel required>
        Example Select
        <FieldLabelInfo title='with extra info in a tooltip' />
      </FieldLabel>
    </FieldRow>
    <FieldRow>
      <Select
        options={[
          ['1', 'item 1'],
          ['2', 'item 2'],
          ['3', 'item 3'],
          ['4', 'item 4'],
          ['5', 'item 5'],
          ['6', 'item 6'],
          ['7', 'item 7'],
          ['8', 'item 8'],
          ['9', 'item 9'],
          ['10', 'item 10'],
        ]}
      />
    </FieldRow>

    <FieldRow>
      <FieldDescription>
        You can select a single option from a list of options
      </FieldDescription>
      <FieldError>You failed to enter a valid value</FieldError>
    </FieldRow>
    <FieldRow>
      <FieldHint>This should help the user enter a valid value</FieldHint>
      <FieldLink href='#'>Link to more information</FieldLink>
    </FieldRow>
  </Field>
);
