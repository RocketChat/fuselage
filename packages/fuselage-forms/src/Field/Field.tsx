import { Field as FieldComponent } from '@rocket.chat/fuselage';
import { ComponentProps, ReactNode } from 'react';

import FieldProvider from './FieldProvider';

type FieldProps = { children: ReactNode } & ComponentProps<
  typeof FieldComponent
>;

function Field({ children, ...props }: FieldProps) {
  return (
    <FieldProvider>
      <FieldComponent {...props}>{children}</FieldComponent>
    </FieldProvider>
  );
}

export default Field;
