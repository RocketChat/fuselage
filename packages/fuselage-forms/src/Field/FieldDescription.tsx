import { FieldDescription as FieldDescriptionComponent } from '@rocket.chat/fuselage';
import type { ReactNode, ComponentProps } from 'react';

import { useFieldDescriptorId } from './FieldContext';

type FieldDescriptionProps = { children: ReactNode } & ComponentProps<
  typeof FieldDescriptionComponent
>;

const FieldDescription = ({ children, ...props }: FieldDescriptionProps) => {
  const id = useFieldDescriptorId('description');

  return (
    <FieldDescriptionComponent {...props} id={id}>
      {children}
    </FieldDescriptionComponent>
  );
};

export default FieldDescription;
