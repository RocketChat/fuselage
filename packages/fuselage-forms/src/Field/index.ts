export { default as Field } from './Field';
export { default as FieldProvider } from './FieldProvider';
export { default as FieldDescription } from './FieldDescription';
export { default as FieldError } from './FieldError';
export { default as FieldHint } from './FieldHint';

export * from './FieldContext';

export * from './Label';

export {
  FieldGroup,
  FieldRow,
  FieldLink,
  FieldLabelInfo,
} from '@rocket.chat/fuselage';
