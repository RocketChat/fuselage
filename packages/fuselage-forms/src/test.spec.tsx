import { composeStories } from '@storybook/react-webpack5';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import * as _stories from './Field/Field.stories';

// TODO: Fix select a11y violations
// "Interactive controls must not be nested (nested-interactive)"
// This is an issue in the component itself and not with the fields
const { WithSelect: _, ...stories } = _stories;

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName || 'Story',
  Story,
]);

test.each(testCases)(
  `renders %s without crashing`,
  async (_storyname, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  },
);

test.each(testCases)(
  '%s should have no a11y violations',
  async (_storyname, Story) => {
    const { container } = render(<Story />);

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  },
);
