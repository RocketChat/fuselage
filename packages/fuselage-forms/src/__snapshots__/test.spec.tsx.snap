// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WithCheckbox without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__label rcx-label"
        >
          Example Checkbox
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
            >
              with extra info in a tooltip
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="with extra info in a tooltip"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </span>
        <label
          class="rcx-box rcx-box--full rcx-check-box"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            Example Checkbox with extra info in a tooltip  *
          </div>
          <input
            aria-describedby="react-aria-:r7:-description react-aria-:r7:-error react-aria-:r7:-hint"
            aria-invalid="true"
            class="rcx-box rcx-box--full rcx-check-box__input"
            id="react-aria-:r7:-label"
            type="checkbox"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-check-box__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r7:-description"
      >
        You can select multiple options
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r7:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r7:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithEmailInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r1:-label"
      >
        Example Email Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r1:-description"
      >
        This field requires a valid email address
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:r1:-description react-aria-:r1:-error react-aria-:r1:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-email rcx-input-box"
          id="react-aria-:r1:-label"
          size="1"
          type="email"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r1:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r1:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithNumberInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r9:-label"
      >
        Example Number Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r9:-description"
      >
        This field requires a valid number
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:r9:-description react-aria-:r9:-error react-aria-:r9:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-number rcx-input-box"
          id="react-aria-:r9:-label"
          size="1"
          type="number"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r9:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r9:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithPasswordInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r2:-label"
      >
        Example Password Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r2:-description"
      >
        This field requires a valid password
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <label
          class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-box--animated rcx-input-box__wrapper"
        >
          <input
            aria-describedby="react-aria-:r2:-description react-aria-:r2:-error react-aria-:r2:-hint"
            aria-invalid="true"
            class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box--type-password rcx-input-box"
            id="react-aria-:r2:-label"
            size="1"
            type="password"
          />
          <span
            class="rcx-box rcx-box--full rcx-input-box__addon"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-eye-off rcx-icon rcx-css-1bepdyv"
            >
              
            </i>
          </span>
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r2:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r2:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithRadioButton without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__label rcx-label"
        >
          Example Radio Button
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
            >
              with extra info in a tooltip
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="with extra info in a tooltip"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </span>
        <label
          class="rcx-box rcx-box--full rcx-radio-button"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            Example Radio Button with extra info in a tooltip  *
          </div>
          <input
            aria-describedby="react-aria-:r5:-description react-aria-:r5:-error react-aria-:r5:-hint"
            aria-invalid="true"
            class="rcx-box rcx-box--full rcx-radio-button__input"
            id="react-aria-:r5:-label"
            type="radio"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-radio-button__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r5:-description"
      >
        You can only select a single option
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r5:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r5:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithSearchInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r3:-label"
      >
        Example Search Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r3:-description"
      >
        You can use this field to search for things
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:r3:-description react-aria-:r3:-error react-aria-:r3:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-search rcx-input-box"
          id="react-aria-:r3:-label"
          size="1"
          type="search"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r3:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r3:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithTelephoneInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r8:-label"
      >
        Example Telephone Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r8:-description"
      >
        This field requires a valid telephone number
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:r8:-description react-aria-:r8:-error react-aria-:r8:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-tel rcx-input-box"
          id="react-aria-:r8:-label"
          size="1"
          type="tel"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r8:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r8:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithTextArea without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r4:-label"
      >
        Example Text Area
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r4:-description"
      >
        You can use this field to enter multi-line text
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <textarea
          aria-describedby="react-aria-:r4:-description react-aria-:r4:-error react-aria-:r4:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-textarea rcx-input-box"
          cols="1"
          id="react-aria-:r4:-label"
          size="1"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r4:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r4:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithTextInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:r0:-label"
      >
        Example Text Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r0:-description"
      >
        Text inputs are used to enter a single line of text.
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:r0:-description react-aria-:r0:-error react-aria-:r0:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box"
          id="react-aria-:r0:-label"
          size="1"
          type="text"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r0:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r0:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithToggleSwitch without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__label rcx-label"
        >
          Example Toggle Switch
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
            >
              with extra info in a tooltip
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="with extra info in a tooltip"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </span>
        <label
          class="rcx-box rcx-box--full rcx-toggle-switch"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            Example Toggle Switch with extra info in a tooltip  *
          </div>
          <input
            aria-describedby="react-aria-:r6:-description react-aria-:r6:-error react-aria-:r6:-hint"
            aria-invalid="true"
            class="rcx-box rcx-box--full rcx-toggle-switch__input"
            id="react-aria-:r6:-label"
            type="checkbox"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:r6:-description"
      >
        This field represents a boolean value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:r6:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:r6:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`renders WithUrlInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-field__label rcx-label"
        for="react-aria-:ra:-label"
      >
        Example URL Input
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
          >
            with extra info in a tooltip
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="with extra info in a tooltip"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
        id="react-aria-:ra:-description"
      >
        This field requires a valid URL
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="react-aria-:ra:-description react-aria-:ra:-error react-aria-:ra:-hint"
          aria-invalid="true"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-url rcx-input-box"
          id="react-aria-:ra:-label"
          size="1"
          type="url"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
        id="react-aria-:ra:-error"
      >
        You failed to enter a valid value
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
          id="react-aria-:ra:-hint"
        >
          This should help the user enter a valid value
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link to more information
        </a>
      </span>
    </div>
  </div>
</body>
`;
